vernac/lemmas.ml

```diff
@@ -207,12 +207,8 @@ let save ?export_seff id const uctx do_guard (locality,poly,kind) hook universes
 
 let default_thm_id = Id.of_string "Unnamed_thm"
 
-let fresh_name_for_anonymous_theorem ~pstate =
-  let avoid = match pstate with
-  | None -> Id.Set.empty
-  | Some pstate -> Id.Set.of_list (Proof_global.get_all_proof_names pstate)
```

Comment:
Note that we drop support for nested `Goal`, I think this is OK.

Reply:
I am OK with it, but following policy in previous PRs this may need to be submitted separately ?

cc: @maximedenes